doc/source/specification.rst

```diff
@@ -1,11 +1,96 @@
 Specification
 =============
 Description of how MorphIO sees morphologies. MorphIO supports :ref:`.asc <specification-neurolucida>` (Neurolucida),
-`.h5`_ (H5 version 1) and `.swc`_ extensions.
+`.h5`_ (H5 version 1) and `.swc`_ extensions. A morphology is represented as a soma with neurites. Either soma or
+neurites can be absent. Neurite is a tree of sections. Section is composed of segments where segment is made of points.
+Lets start describing them from the smallest to the biggest entity.
 
-Soma formats:
--------------
-MorphIO implements Soma specification from `NeuroMorpho`_, and recognizes several kinds of soma format.
+
+Point
+-----
+A point is an array of three numbers **[X, Y, Z]** where X, Y, Z are Cartesian coordinates of position.
+A point always has a diameter associated with it.
+
+
+Segment
+-------
+A segment consists of two consecutive points(`Point`_) belonging to the same section(`Section`_).
+
+.. image:: images/swc/two_pt.svg
+   :scale: 100 %
+   :alt: segment
+
+
+Section
+-------
+
+A section is a series of one or more segments. Each section has a type associated with it. The type shows what part of
+the neuron a section represents. The type can of the following: axon, soma, basal dendrite, apical dendrite or
+custom dendrite. `Section API <cpp/doxygen_Section.html>`_.
+
+.. image:: images/section_segment.svg
+   :scale: 100 %
+   :alt: section
+
+The first and the last point of section must be of the following combinations:
+
+1. starting point, forking point
+2. forking point, end point
+3. forking point, forking point
+4. starting point, end point
+
+Starting point is the first point of a section without parent. Forking point is the last point of a section that has
+multiple children. End point is the last point of a section without children. For more details on section's parent and
+children see `Neurite`_. The above combinations are marked with their corresponding number at the image below.
+
+.. image:: images/sections.svg
+   :scale: 100 %
+   :alt: section variants
+
+Neurite
+-------
+
+A neurite is a tree of sections(`Section`_). The tree structure implies the following:
+
+* Section can have only one parent (another section).
+* Section can have an arbitrary number of children (other sections).
+* No loops are present in the structure.
```

Comment:
```suggestion
* It exists a root section for the neurite.
* The structure is oriented : all its edges point away from the root.
* No cycles are present in the structure.
```

(In graph theory, a loop is a special case of cycles where the edge loop on the same node. Also called the self-loop)